src/mono/msbuild/android/build/AndroidBuild.targets

```diff
@@ -94,7 +93,7 @@
     <PropertyGroup>
       <_AOTMode Condition="'$(UseMonoJustInterp)' != 'true'">Normal</_AOTMode>
       <_AOTMode Condition="'$(UseMonoJustInterp)' == 'true'">JustInterp</_AOTMode>
-      <_AOTMode Condition="'$(ForceFullAOT)' == 'true'">Full</_AOTMode>
+      <_AOTMode Condition="'$(ForceAOT)' == 'true' and '$(AOTWithLibraryFiles)' != 'true'">Full</_AOTMode>
```

Comment:
@steveisok not sure if this is correct but I noticed we were using an undocumented property `ForceFullAOT` (I found its usage only on your library sample repo). 
So I switched it to this combination, maybe it needs a refinement